packages/flutter_tools/lib/src/ios/ios_project_migration.dart

```diff
@@ -0,0 +1,141 @@
+      if (line.contains('3B80C3941E831B6300D905FE')) {
+        return null;
+      }
+
+      // App.framework Embed Framework reference (build phase to embed framework).
+      // 3B80C3951E831B6300D905FE /* App.framework in Embed Frameworks */,
+      if (line.contains('3B80C3951E831B6300D905FE')
+          || line.contains('741F496821356857001E2961')) { // Ephemeral add-to-app variant.
+        return null;
+      }
+
+      // App.framework project file reference (seen in Xcode navigator pane).
+      // isa = PBXGroup;
+      // children = (
+      //	 3B80C3931E831B6300D905FE /* App.framework */,
+      if (line.contains('3B80C3931E831B6300D905FE')
+          || line.contains('741F496521356807001E2961')) { // Ephemeral add-to-app variant.
+        return null;
+      }
+
+      // Flutter.framework Frameworks reference.
+      // isa = PBXFrameworksBuildPhase;
+      // files = (
+      //   9705A1C61CF904A100538489 /* Flutter.framework in Frameworks */,
+      if (line.contains('9705A1C61CF904A100538489')) {
+        return null;
+      }
+
+      // Flutter.framework Embed Framework reference (build phase to embed framework).
+      // 9705A1C71CF904A300538489 /* Flutter.framework in Embed Frameworks */,
+      if (line.contains('9705A1C71CF904A300538489')
+          || line.contains('741F496221355F47001E2961')) { // Ephemeral add-to-app variant.
+        return null;
+      }
+
+      // Flutter.framework project file reference (seen in Xcode navigator pane).
+      // isa = PBXGroup;
+      // children = (
+      //	 9740EEBA1CF902C7004384FC /* Flutter.framework */,
+      if (line.contains('9740EEBA1CF902C7004384FC')
+          || line.contains('741F495E21355F27001E2961')) { // Ephemeral add-to-app variant.
+        return null;
+      }
+
+      // Embed and thin frameworks in a script instead of using Xcode's link / embed build phases.
+      const String thinBinaryScript = 'xcode_backend.sh\\" thin';
+      if (line.contains(thinBinaryScript)) {
+        return line.replaceFirst(thinBinaryScript, 'xcode_backend.sh\\" embed_and_thin');
+      }
+
+      return line;
+    });
+  }
+
+  void _migratePodfile() {
+    if (!_podfile.existsSync()) {
+      _logger.printTrace('Podfile not found, skipping migration');
+      return;
+    }
+
+    _processFileLines(_podfile, (String line) {
+      // # Prevent Cocoapods from embedding a second Flutter framework and causing an error with the new Xcode build system.
+      // install! 'cocoapods', :disable_input_output_paths => true
+      if (line.contains('Prevent Cocoapods from embedding a second Flutter framework and causing an error with the new Xcode build system.')) {
+        return null;
+      }
+      if (line.contains('install! \'cocoapods\', :disable_input_output_paths => true')) {
+        return null;
+      }
+
+      return line;
+    });
+  }
+
+  void _processFileLines(File file, _ProjectFileLineProcessor processLine) {
```

Comment:
Since this is a private member with a private typedef, I would instead do the function type inline:

```
void _processFileLines(File file, String Function(String) processLine)
```